### DISCUSSION TITLE
How to have different bindings in the same application

### DISCUSSION THREAD

**Discussion by JonDehn** [Textualize/textual#5399] [Q&A]
I have an application I'd like to run in two "modes" (specified by a program parameter).  The set of bindings would be different depending on the mode (think "read only" mode vs. "modify" mode).  I see where one can disable bindings use check_actions, but they still show up (greyed out) in the tool bar.  Before launching the class based on App, I'd like to set the bindings.  I suspect I just don't know enough about how to do this; a simple setting of the class-wide variable "BINDINGS" after the App object is created gives a "FrozenInstanceError".

**Comment by darrenburns** [comment]
You can add define a keymap, and conditionally load in the keymap you want depending on the program mode: https://darren.codes/posts/textual-keymaps/

**Comment by JonDehn** [comment]
Thanks for your quick reply.  I looked at your reference, but I don't see how that can change the number of keys displayed in the footer.  
In my case, if the application is started in "read only" mode, perhaps I'd like a BINDING with one thing in it. Binding('B', 'browse', 'browse file').   If started in read/write mode, then perhaps three keys: Binding('B', 'browse', browse file'), Binding('D', 'delete', 'delete file'), Binding('A', 'add', 'add file').   The example you cite simply changed the keys that would be bound to the existing actions.

**Reply by darrenburns** [reply]
Oh, I see. You might literally be looking for "modes" then 😄 

https://textual.textualize.io/guide/screens/#modes

Does that seem like what you need?

**Reply by darrenburns** [reply]
Or if your "modes" are too similar to justify that, perhaps returning `False` from `check_action` to hide them from the footer? https://textual.textualize.io/api/dom_node/#textual.dom.DOMNode.check_action

**Comment by JonDehn** [comment]
My modes are very similar, so I will investigate your hint of returning False from check_action.  Thanks
